src/drake/core.clj

```diff
@@ -176,6 +181,23 @@
       :vars vars
       :opts (if-not method opts (merge (:opts method) opts)))))
 
+(defn- make-input-stats
```

Comment:
I can see the case for generating these structures at parse time (similar to the "two-copies" approach Artem suggested in his comment about this feature), but honestly the parser code confuses me so I did this here.

We'd save a couple extra checks for existence, but the change will be more extensive as inputs will now be this map instead of just a string, and everything has to handle that. It's your call.